onvm/onvm_nflib/onvm_common.h

```diff
@@ -495,4 +496,32 @@ whether_wakeup_client(struct onvm_nf *nf, struct nf_wakeup_info *nf_wakeup_info)
 
 #define RTE_LOGTYPE_APP RTE_LOGTYPE_USER1
 
+/*
+ * Updates the ether_addr struct with a fake, safe MAC address.
+ */
+static inline int
+onvm_get_fake_macaddr(struct ether_addr *mac_addr) {
+        uint16_t *mac_addr_bytes = (uint16_t *)((struct ether_addr *)(mac_addr)->addr_bytes);
+        mac_addr_bytes[0] = 2;
+        mac_addr_bytes[1] = 0;
+        mac_addr_bytes[2] = 0;
+        return 0;
+}
+
+/*
+ * Tries to fetch the MAC address of the port_id. 
+ * Uses fake MAC address if port_id is invalid.
+ * Return 0 if port is valid, -1 if port is invalid.
+ */
+static inline int
+onvm_get_macaddr(uint8_t port_id, struct ether_addr *mac_addr) {
+        if (rte_eth_dev_is_valid_port(port_id)) {
+                rte_eth_macaddr_get(port_id, mac_addr);
+                return 0;
+        } else {
+                onvm_get_fake_macaddr(mac_addr);
```

Comment:
I think that we can remove this part of the function. My idea is that onvm_get_macaddr is just checking for a valid port and fetching the macaddr for that port on success, return -1 otherwise. I don't think we should call onvm_get_fake_macaddr here. We should modify the NF's to decide whether or not they need a fake mac addr on their own.

Reply:
Oh okay got it. So, speed_tester and scaling should use `onvm_get_fake_macaddr`? But they do take in a dst MAC in which case it would be good to try to get a real address for the source but can settle for a fake address. Or do we go along the lines that if a fake address is sufficient, then we needn't bother to fetch the real address?

Reply:
I think the qn is whether returning the fake MAC address is an extra burden to be vary of. Otherwise, the return value denotes the status and the user can decide what to do with the address they received.